consensus/src/quorum_store/utils.rs

```diff
@@ -67,7 +67,7 @@ impl BatchBuilder {
             });
             self.data.push(serialized_txn);
 
-            true
+            self.num_txns < self.max_txns && self.num_bytes < self.max_bytes
```

Comment:
Are not we checking it in the if in line 59?

Reply:
nope. imagine fragment size = 200 txns and batch size = 200 txns. You will only realize you are over the max batch when you form the next fragment, so you always end up forming one more fragment if your batch size is perfect multiplier of fragment size.